cmd/runmqserver/qmgr.go

```diff
@@ -86,37 +86,25 @@ func configureQueueManager() error {
 		log.Println(err)
 		return err
 	}
-
 	for _, file := range files {
 		if strings.HasSuffix(file.Name(), ".mqsc") {
 			abs := filepath.Join(configDir, file.Name())
 			// #nosec G204
 			cmd := exec.Command("runmqsc")
-			stdin, err := cmd.StdinPipe()
-			if err != nil {
-				log.Println(err)
-				return err
-			}
-			// Open the MQSC file for reading
-			// #nosec G304
-			f, err := os.Open(abs)
-			if err != nil {
-				log.Printf("Error opening %v: %v", abs, err)
-			}
-			// Copy the contents to stdin of the runmqsc process
-			_, err = io.Copy(stdin, f)
-			if err != nil {
-				log.Errorf("Error reading %v: %v", abs, err)
-			}
-			err = f.Close()
+			// Read mqsc file into variable
+			mqsc, err := ioutil.ReadFile(abs)
 			if err != nil {
-				log.Errorf("Failed to close MQSC file handle: %v", err)
+				log.Printf("Error reading file %v: %v", abs, err)
 			}
-			err = stdin.Close()
+			// Write mqsc to buffer
+			var buffer bytes.Buffer
+			_, err = buffer.Write(mqsc)
 			if err != nil {
-				log.Errorf("Failed to close MQSC stdin: %v", err)
+				log.Printf("Error writing mqsc file %v to buffer: %v", abs, err)
 			}
-			// Run the command and wait for completion
+			// Buffer mqsc to stdin of runmqsc
+			cmd.Stdin = &buffer
+			// Run runmqsc command
 			out, err := cmd.CombinedOutput()
 			if err != nil {
```

Comment:
I think the "out" statement should be under an "else", so that we either print the error or the regular output message.